### ISSUE TITLE
'jsonString' argument missing or empty - Any/all amplify cli commands

### ISSUE THREAD

**Issue by evanlarkin10** [aws-amplify/amplify-cli#8471]
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

v14.13.0

### Amplify CLI Version

Cant run this command

### What operating system are you using?

mac

### Amplify Categories

auth, storage, analytics

### Amplify Commands

env, init, status

### Describe the bug

I all of a sudden can't run any amplify commands without this error output, not even amplify version. Happened randomly after coming back to a codebase with no changes after using amplify cli on the project it without issues the day before.

### Expected behavior

amplify cli allows me to run commands without error.

### Reproduction steps

npm install -g @aws-amplify/cli
amplify status // amplify version // amplify checkout env // amplify <anything>

### Log output

<details>

```
EL-MacBook-Pro:dinnerbell elarkin$ amplify status
'jsonString' argument missing or empty
Error: 'jsonString' argument missing or empty
    at Function.JSONUtilities.parse (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:79:15)
    at Function.JSONUtilities.readJson (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:47:32)
    at StateManager.getData (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:282:56)
    at StateManager.getLocalEnvInfo (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:91:25)
    at Function.HooksMeta.getInstance (/usr/local/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/hooks/hooksMeta.js:99:76)
    at Object.run (/usr/local/lib/node_modules/@aws-amplify/cli/lib/index.js:138:56)


```

</details>

### Additional information

I made no code changes, just wanted to change amplify environment and now I am getting this error with every amplify command. I tried deleting node modules and reintalling them and reinstalling aws-amplify cli. 

Error: 'jsonString' argument missing or empty

**Comment by marcvberg** [comment]
if you have both directories available - is there any difference between `amplify/cli.json` in the two directories?

**Comment by loganpowell** [comment]
I just updated to the latest CLI and now I'm getting this when I try to run any command:

```sh
amplify --version
'jsonString' argument missing or empty
Error: 'jsonString' argument missing or empty
    at Function.JSONUtilities.parse (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:79:15)
    at Function.JSONUtilities.readJson (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/jsonUtilities.js:47:32)
    at StateManager.getData (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:298:56)
    at StateManager.getMeta (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:42:31)
    at getSafeAccountId (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/lib/context-manager.js:53:61)
    at attachUsageData (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/lib/context-manager.js:47:23)
    at Object.run (/Users/logan/.nvs/node/14.18.1/x64/lib/node_modules/@aws-amplify/cli/lib/index.js:147:53)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

I believe it may be an issue that is related to [recommended `.gitignore` settings](https://github.com/aws-amplify/amplify-cli/issues/4700#issuecomment-655650523) containing `amplify-meta.json`.

Any tips?

**Comment by loganpowell** [comment]
@josefaidt thank you for attending to this, but now that it's been labeled as a `feature-request`, does that mean we shouldn't expect it to be fixed in the near term? Do you have any ideas what I may be doing wrong on my end?

**Comment by josefaidt** [comment]
Hey @loganpowell :wave: this is labeled as a feature request to improve the error messages we receive when we run into something like this. In the mean time, if you are missing the `amplify/cli.json` file

**Comment by loganpowell** [comment]
Hey @josefaidt! Thanks for the tips. I actually just deleted the entire repo from my local machine (after the global install of the new CLI) and re cloned it. I then ran the `amplify pull --appId <appid> --envName <env>` command and now it works!

**Comment by josefaidt** [comment]
Ahh glad to here you're back up and running!

> It seems that the old CLI configs cause some conflicts with the new CLI 🤷

What sort of issues are you seeing?

**Comment by loganpowell** [comment]
> What sort of issues are you seeing?

Just the titular error: `'jsonString' argument missing or empty`

**Comment by josefaidt** [comment]
Hey @loganpowell thanks for confirming. Does the `'jsonString'...` error your seeing now have the same stack trace? Are you seeing the `amplify/backend/amplify-meta.json` file in your project?

**Comment by loganpowell** [comment]
Hi @josefaidt. Thank you for being so courteous. I don't get the same `jsonString` error since I deleted the repo and recloned, but now I am getting a new error when I try to `amplify push`. 

I've submitted another [issue](https://github.com/aws-amplify/amplify-cli/issues/9322) for that though...

**Comment by josefaidt** [comment]
Ahh that file is hidden by the editor-specific settings generated by the CLI. Could that be the case as to why you don't see it?

I'll take a look at the other issue as well!

**Comment by loganpowell** [comment]
yes, [the recommendation](https://github.com/aws-amplify/amplify-cli/issues/4700#issuecomment-653760052) used to be to add it to the `.gitignore` manually, but it seems that the new CLI adds it there automatically.

**Comment by josefaidt** [comment]
You are correct! That file can contain sensitive information and is generated by the CLI when we initialize an environment

**Comment by loganpowell** [comment]
@josefaidt got that second issue resolved as well. 🙏  thanks again!

**Comment by josefaidt** [comment]
Ahh awesome! Glad to hear you're back up and running @loganpowell 🙌  I'll leave this issue open to improve the logging for when one of these critical files are missing and fail gracefully 🙂

**Comment by YazidHamdi** [comment]
I have also just had the same issue and I think I know what's happening: when you have created a function that has secret values in one environment (say "dev") and it's pushed and all good, then you checkout "prod" for example and instead of running "amplify push" you run "amplify publish" for some reason the CLI doesn't ask about the secret's value for this new environment, which is what breaks the publish with the reported error message.
To summarize: "amplify publish" needs to do the same prompting (see attached) in case functions with unconfigured secrets are being pushed
![Screenshot 2022-03-04 161448](https://user-images.githubusercontent.com/16480893/156789173-4e50f546-6c1b-4d3a-8a61-92ff73c0f15f.png)
.

**Comment by radumetawealth** [comment]
I have solved this by deleting amplify-meta.json file.

I'm using an M1 Pro mac.
